### PR TITLE
Fix/null pointer exception

### PR DESCRIPTION
**Pull request by bxy4543**
### Describe what this PR does / why we need it

null pointer exception

### CODE REVIEW

File: apply/current_cluster.go [hunk 1]
```diff
@@ -41,7 +41,7 @@ func listNodes() *corev1.NodeList {
 	}
 	nodes, err := client.ListNodes(c)
 	if err != nil {
-		logger.Info("current cluster nodes not found, will create a new cluster")
+		logger.Info("current cluster nodes not found, will create a new cluster %v", err)
```
Comment (allencloud) [inline apply/current_cluster.go:44]: Why to use an Info level?

File: apply/current_cluster.go [hunk 2]
```diff
@@ -96,6 +96,9 @@ func deleteNode(name string) error {
 func DeleteNodes(nodeIPs []string) error {
 	logger.Info("delete nodes %s", nodeIPs)
 	nodes := listNodes()
+	if nodes == nil {
+		return fmt.Errorf("failed to get node list")
```
Comment (allencloud) [inline apply/current_cluster.go:100]: failed to get node list: no nodes found

We could show error message as explicit as possible.

File: apply/current_cluster.go (hunk 1 above)
Comment (fanux) [reply apply/current_cluster.go:44]: When we create a new cluster, the listNodes will return error because current cluster is not exist, but it's normal.

**Review by fanux — APPROVED** [review]

File: apply/current_cluster.go [hunk 3]
```diff
@@ -33,23 +33,22 @@ func GetCurrentCluster() (*v1.Cluster, error) {
...
 	if err != nil {
-		logger.Info("current cluster not found, will create a new cluster %v", err)
-		return nil
+		return nil, fmt.Errorf("current cluster not found, will create a new cluster %v", err)
 	}
 	nodes, err := client.ListNodes(c)
 	if err != nil {
-		logger.Info("current cluster nodes not found, will create a new cluster")
-		return nil
+		return nil, fmt.Errorf("current cluster nodes not found, will create a new cluster %v", err)
```
Comment (fanux) [inline apply/current_cluster.go:43]: will create a new cluster should not in this function. write it in get current nodes

**Review by fanux — APPROVED** [review]

**Pull request merged**